### PR TITLE
Login & automatic membership system

### PR DESCRIPTION
**Pull request by mfaalk**
MOVED TO PR #109

### CODE REVIEW

File: login.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
+<?php
+include('config/config.php');
+if ($enableLogin === true) {
+    if (isset($_POST['submit_updatePwd'])) {
+        if (!empty($_POST["password"]) && ($_POST["password"] == $_POST["repassword"])) {
+            $passwordErr = '';
+            $password = $_POST["password"];
+            
+            if (strlen($_POST["password"]) <= '5') {
```
Comment (hammydown4325) [inline login.php:70]: Doesnt make sense. Says Less than or equal to 5 and the error says 8?

File: pre-index.php [hunk 2]
```diff
@@ -848,6 +877,35 @@ class="onoffswitch-checkbox"/>
...
             </center>
         </div>
+		<?php
+        if ($enableLogin === true && $_SESSION['user']->email) {
+            ?>
+			<div>
+				<center>
+					<button class="settings"
+							onclick="document.location.href='logout.php'">
+						<i class="fa" aria-hidden="true"></i> Logout
```
Comment (hammydown4325) [inline pre-index.php]: Needs to be translatable

File: pre-index.php [hunk 3]
```diff
@@ -848,6 +877,35 @@ class="onoffswitch-checkbox"/>
...
+			</div><br>
+		    <div>
+                <center>
+                    <p>
+                    <?php
+                    $time = date("Y-m-d", $_SESSION['user']->expire_timestamp);
+                    
+                    echo $_SESSION['user']->email . "<br>";
+                    if ($_SESSION['user']->expire_timestamp > time()) {
+                        echo "<span style='color: green;'>Account expires on {$time}</span>";
```
Comment (hammydown4325) [inline pre-index.php]: Needs to be translatable

File: pre-index.php [hunk 4]
```diff
@@ -848,6 +877,35 @@ class="onoffswitch-checkbox"/>
...
+                    } else {
+                        echo "<span style='color: green;'>Account expired on {$time}</span>";
```
Comment (hammydown4325) [inline pre-index.php]: Needs to be translatable

File: login.php [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            $password = $_POST["password"];
+            
+            if (strlen($_POST["password"]) <= '5') {
+                $passwordErr = "<b>Your password must contain at least 8 characters!</b><br>";
+            } elseif (!preg_match("#[0-9]#", $password)) {
+                $passwordErr = "<b>Your password must contain at least 1 number!</b><br>";
+            } elseif (!preg_match("#[A-Z]#", $password)) {
+                $passwordErr = "<b>Your password must contain at least 1 capital letter!</b><br>";
+            } elseif (!preg_match("#[a-z]#", $password)) {
+                $passwordErr = "<b>Your password must contain at least 1 lowercase letter!</b><br>";
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

File: login.php [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        if (!empty($_POST["password"]) && ($_POST["password"] == $_POST["repassword"])) {
+            $passwordErr = '';
+            $password = $_POST["password"];
+            
+            if (strlen($_POST["password"]) <= '5') {
+                $passwordErr = "<b>Your password must contain at least 8 characters!</b><br>";
+            } elseif (!preg_match("#[0-9]#", $password)) {
+                $passwordErr = "<b>Your password must contain at least 1 number!</b><br>";
+            } elseif (!preg_match("#[A-Z]#", $password)) {
+                $passwordErr = "<b>Your password must contain at least 1 capital letter!</b><br>";
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

File: login.php [hunk 7]
```diff
@@ -0,0 +1,98 @@
+<?php
+include('config/config.php');
+if ($enableLogin === true) {
+    if (isset($_POST['submit_updatePwd'])) {
+        if (!empty($_POST["password"]) && ($_POST["password"] == $_POST["repassword"])) {
+            $passwordErr = '';
+            $password = $_POST["password"];
+            
+            if (strlen($_POST["password"]) <= '5') {
+                $passwordErr = "<b>Your password must contain at least 8 characters!</b><br>";
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

File: login.php [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+if ($enableLogin === true) {
+    if (isset($_POST['submit_updatePwd'])) {
+        if (!empty($_POST["password"]) && ($_POST["password"] == $_POST["repassword"])) {
+            $passwordErr = '';
+            $password = $_POST["password"];
+            
+            if (strlen($_POST["password"]) <= '5') {
+                $passwordErr = "<b>Your password must contain at least 8 characters!</b><br>";
+            } elseif (!preg_match("#[0-9]#", $password)) {
+                $passwordErr = "<b>Your password must contain at least 1 number!</b><br>";
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

File: login.php [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+            } elseif (!preg_match("#[A-Z]#", $password)) {
+                $passwordErr = "<b>Your password must contain at least 1 capital letter!</b><br>";
+            } elseif (!preg_match("#[a-z]#", $password)) {
+                $passwordErr = "<b>Your password must contain at least 1 lowercase letter!</b><br>";
+            }
+        } else {
+            $passwordErr = "<b>Your passwords didn't match!</b><br>";
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

File: login.php [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+            $_SESSION['user']->updatePwd = $info['updatePwd'];
+            if ($info['updatePwd'] == 1) {
+                header("Location: /login.php");
+                die();
+            } else {
+                header("Location: .");
+                die();
+            }
+        } else {
+            echo "Incorrect username or password.";
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

File: login.php [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+        }
+    }
+
+    if ($_SESSION['user']->updatePwd == 1) {
+        ?>
+		Please change your password.
+		<form action='' method='POST'>
+			<table>
+				<tr>
+					<th>New password</th><td><input type="password" name="password" required></td>
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

File: login.php [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+				</tr>
+				<tr>
+					<th>Confirm password</th><td><input type="password" name="repassword" required></td>
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

File: login.php [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+				</tr>
+			</table>
+		</form>
+   <?php
+    } else {
+        ?>
+		<form action='' method='POST'>
+			<table>
+				<tr>
+					<th>E-mail</th><td><input type="text" name="email" required></td>
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

File: login.php [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+				</tr>
+				<tr>
+					<th>Password</th><td><input type="password" name="password" required></td>
```
Comment (hammydown4325) [inline login.php]: Needs to be translatable

**Review by hammydown4325 — COMMENTED** [review]
Initial Review. More to come

File: login.php (hunk 1 above)
Comment (mfaalk) [reply login.php:70]: Solved. Forgot to update the error message. Min character set to 6.

**Comment by mfaalk** [comment]
IV is still sent in raw_data so I'm thinking of setting $noHighLevelData = true, minIV = '0', minLevel= '0' in config for non logged on/paying users to solve that issue.

**Comment by Glennmen** [comment]
I see you are using SHA1, wouldn't it be better to use something more secure?

**Comment by mfaalk** [comment]
Sure. I haven't done much web development in a couple of years so I just did a quick google search and found sha1.
password_hash may be more interesting?

Also, I haven't added anything for sql injections, I figured the framework would take care of that?

**Comment by Glennmen** [comment]
@mfaalk PHP docs have lots of info: http://php.net/manual/en/function.password-hash.php

You mean the Medoo framework? If you use the prepared statements then it should.

**Comment by mfaalk** [comment]
I messed up.
Opening a second PR.

**Pull request closed without merging**